src/hazelcore/semantics/UHPat.re

```diff
@@ -167,3 +167,32 @@ let text_operand =
       u_gen,
     );
   };
+
+let rec is_complete = (pn: t): bool => {
+  switch (pn) {
+  | OpSeq(Placeholder(n) as _skel, seq) =>
+    is_complete_operand(seq |> Seq.nth_operand(n))
+  | OpSeq(BinOp(InHole(_), _, _, _), _) => false
+  | OpSeq(BinOp(NotInHole, _, skel1, skel2), seq) =>
+    is_complete(OpSeq(skel1, seq)) && is_complete(OpSeq(skel2, seq))
```

Comment:
you should define an auxiliary function `is_complete_skel` that takes a skel and a seq, rather than constructing intermediate OpSeqs (which are invalid as standalone OpSeqs)

Reply:
fixed